### DISCUSSION TITLE
Unique Id Error for Chrome Cast Device

### DISCUSSION THREAD

**Discussion by Happy-Cadaver** [music-assistant/support#3256] [Q&A] [answered]
I'm running 2024.12.0 beta and trying out the MA native integration, I like it so far and it's inspired me to finally use HA for music. I added, as a test subject, a Google Nest Mini to my player providers which seems to be working fine. However whenever I restart HA I get this in my logs.
`2024-12-03 21:12:39.694 ERROR (MainThread) [homeassistant.components.media_player] Platform music_assistant does not generate unique IDs. ID 03a489fa-888c-e412-c376-985042342252 already exists - ignoring media_player.living_room_speaker_2`
Yet there's no `media_player.living_room_speaker_2` in my devices list, just a `media_player.living_room_speaker`. Is this a bug I should report? Or am I just not understanding how to set it up, which is likely to be honest. I just hate seeing errors in my logs and this is the only one left to deal with at the moment.

**Comment by OzGav** [comment]
@jozefKruszynski

**Reply by jozefKruszynski** [reply]
This is a strange one, haven't seen this reported by anyone else using the integration in HA beta.

Gonna tag in @marcelveldt also.

**Reply by jozefKruszynski** [reply]
I have a feeling that this is related to removing the `mass_` prefix from the unique_id, which was done during initial code reviews.
This should be raised as a bug in the HA beta I think, but perhaps @marcelveldt can confirm that.

**Reply by jozefKruszynski** [reply]
Iirc at the time was suggested to remove this as unique IDs are unique within their own domain, however, because we are creating media_player entities and referencing the original player ID, we are indeed not unique.
@marcelveldt Do we add back the prefix?

**Reply by marcelveldt** [reply]
No, that is not related.

**Reply by jozefKruszynski** [reply]
Fair enough, then forget everything I said 🤣

**Reply by marcelveldt** [reply]
wait... yet it is. HA doesn't use the integration domain prefix but the platform domain. That is indeed going to hinder if you add a cast player from HA to MA (and then back to HA) - Its a bit of a long shot because you shouldn't be using the HA player provider for cast players in the first place.

**Reply by jozefKruszynski** [reply]
Do we need to do something to mitigate this?

**Comment by OzGav** [comment]
This will be looked at soon

**Reply by Happy-Cadaver** [reply]
Thank you.

**Comment by radinsky** [comment]
2025.5 still see this, any way to fix?

**Reply by OzGav** [reply, marked as answer]
This hasn’t been mentioned for 1.5 years. Try restarting MA and HA..
